File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        q = deque()
        begin = list(['(', '{', "["])

        for i in s:
            if i in begin:
                q.append(i)
            else:
                if len(q) < 1:
                    return False
                elif i == ')':
                    if q[-1] != '(':
                        return False
                    else:
                        q.pop()
                elif i == '}':
                    if q[-1] != '{':
                        return False
                    else:
                        q.pop()
                elif i == ']':
                    if q[-1] != '[':
                        return False
                    else:
                        q.pop()
        
        if len(q) == 0:
            return True
        else:
            return False
        

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-1.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<pair<int, vector<int>>, vector<pair<int, vector<int>>>,greater<pair<int, vector<int>>>> pq;
        for(const auto& v : points)
        {
            pq.push(make_pair(v[0]*v[0] + v[1]*v[1], v));
        }
        vector<vector<int>> ans;
        for(int i =0; i < k; i++)
        {
            ans.push_back(pq.top().second);
            cout << pq.top().second[0] << " " << pq.top().second[1] << '\n';
            pq.pop();
            
        }
        
        return ans;
    }
};
